linera-views/src/backends/dynamo_db.rs

```diff
@@ -352,28 +370,34 @@ impl AdminKeyValueStore for DynamoDbStoreInternal {
             .map(|n| Arc::new(Semaphore::new(n)));
         let max_stream_queries = config.common_config.max_stream_queries;
         let namespace = namespace.to_string();
-        let root_key = root_key.to_vec();
-        Ok(Self {
+        let start_key = extend_root_key(root_key);
+        let store = Self {
             client,
             namespace,
             semaphore,
             max_stream_queries,
-            root_key,
-        })
+            start_key,
+        };
+        store.write_root_key(root_key).await?;
+        Ok(store)
     }
 
-    fn clone_with_root_key(&self, root_key: &[u8]) -> Result<Self, DynamoDbStoreInternalError> {
+    async fn clone_with_root_key(
+        &self,
+        root_key: &[u8],
+    ) -> Result<Self, DynamoDbStoreInternalError> {
+        self.write_root_key(root_key).await?;
```

Comment:
Does that mean `DbStorage::load_chain` will now write a root key?

E.g. if we handle a certificate only to realize during execution that it's invalid, it would still write that chain state view's root key to storage?

Reply:
Yes.
That is the design with the root keys. The running of the system would still be deterministic.

Reply:
But we're calling `load_chain` quite early in the validation process: You could spam a validator with completely invalid certificates with random bogus chain IDs, and it would create a database entry for each of these IDs.

Reply:
Yes, this sounds much better to delay the registration of a root view to the first `write_batch`.

Reply:
Ok, done.